internal/admin/handle_exports.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package admin
+
+import (
+	"fmt"
+	"net/http"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/google/exposure-notifications-server/internal/export/database"
+	"github.com/google/exposure-notifications-server/internal/export/model"
+)
+
+// HandleExportsSave handles the create/update actions for exports.
+func (s *Server) HandleExportsSave() func(c *gin.Context) {
+	return func(c *gin.Context) {
+		var form exportFormData
+		err := c.Bind(&form)
+		if err != nil {
+			ErrorPage(c, err.Error())
+			return
+		}
+
+		ctx := c.Request.Context()
+		m := TemplateMap{}
+
+		exportDB := database.New(s.env.Database())
+		exportConfig := &model.ExportConfig{}
+		idParam := c.Param("id")
+		if idParam != "0" {
+			cfgID, err := strconv.ParseInt(idParam, 10, 64)
+			if err != nil {
+				ErrorPage(c, "unable to parse `id` param.")
+				return
+			}
+			exportConfig, err = exportDB.GetExportConfig(ctx, cfgID)
+			if err != nil {
+				ErrorPage(c, fmt.Sprintf("Error loading export config: %v", err))
+				return
+			}
+		}
+
+		if err := form.PopulateExportConfig(exportConfig); err != nil {
+			ErrorPage(c, fmt.Sprintf("error processing export config: %v", err))
+		}
+
+		updateFn := exportDB.AddExportConfig
+		if exportConfig.ConfigID != 0 {
+			updateFn = exportDB.UpdateExportConfig
+		}
+		if err := updateFn(ctx, exportConfig); err != nil {
+			ErrorPage(c, fmt.Sprintf("Error writing export config: %v", err))
+			return
+		}
+
+		m.AddSuccess(fmt.Sprintf("Updated export config #%v", exportConfig.ConfigID))
+
+		usedSigInfos := make(map[int64]bool)
+		for _, id := range exportConfig.SignatureInfoIDs {
+			usedSigInfos[id] = true
+		}
+
+		sigInfos, err := exportDB.ListAllSignatureInfos(ctx)
+		if err != nil {
+			ErrorPage(c, fmt.Sprintf("Error reading the database: %v", err))
+		}
+
+		m["export"] = exportConfig
+		m["usedSigInfos"] = usedSigInfos
+		m["siginfos"] = sigInfos
+		c.HTML(http.StatusOK, "export", m)
+	}
+}
+
+// HandleExportsShow handles the show action for exports.
+func (s *Server) HandleExportsShow() func(c *gin.Context) {
+	return func(c *gin.Context) {
+		ctx := c.Request.Context()
+		m := TemplateMap{}
+
+		exportDB := database.New(s.env.Database())
+		exportConfig := &model.ExportConfig{}
+		if idParam := c.Param("id"); idParam == "0" {
```

Comment:
Ditto about `idParam == ""` too?

Reply:
It's required to be a value by the router to reach this function. The router enforces a digit in the path.

This PR is just moving existing code around. To reduce the size of a diff and to reduce the chance of introducing a bug, I'm only moving code, no refactors or other changes in this PR.